### PR TITLE
Move optimsation geometry to the geometry module

### LINKED ISSUE #2414: Move geometry optimisation functions into geometry module
## Description

We should move the `optimisation.geometry` module into the root `geometry` module (i.e. `geometry.optimisation`). The `optimisation` module should be largely standalone, and should not depend on the `geometry`.

This change will also allow us to remove the geometry constraint registry and put the geometry constraints directly on the `GeometryParameterisation` classes.

### PR DESCRIPTION
**Pull request by je-cook**
## Linked Issues

Closes #2414 

## Description

Moves the geometry optimisation to the geometry module. The old API thankfully has no name clashes so that has just been moved into the folder for the time being.

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by hsaunders1904 — COMMENTED** [review]
Looks great - a couple of small questions

File: bluemira/geometry/parameterisations.py [hunk 1]
```diff
@@ -847,40 +823,28 @@ def __init__(self, var_dict: Optional[Dict[str, float]] = None):
         )
         variables.adjust_variables(var_dict, strict_bounds=False)
         super().__init__(variables)
-        self.n_ineq_constraints = 1
 
-    def shape_ineq_constraints(
```
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/geometry/parameterisations.py:852]: yeah probably I'll rework it slightly.

[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/geometry/parameterisations.py:852]: done

File: bluemira/builders/tf_coils.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
 if TYPE_CHECKING:
     from bluemira.geometry.parameterisations import GeometryParameterisation
-    from bluemira.optimisation._geometry.typing import GeomConstraintT
+    from bluemira.geometry.optimisation._typing import GeomConstraintT
```
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/builders/tf_coils.py]: done

**Pull request merged**